### PR TITLE
New Resource: `azurerm_system_center_virtual_machine_manager_virtual_network`

### PR DESCRIPTION
**Pull request by neil-yechenwei**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description

This PR is to support new resource System Center Virtual Machine Manager Virtual Network in SCVMM RP. Maybe these three PRs (https://github.com/hashicorp/terraform-provider-azurerm/pull/25449, https://github.com/hashicorp/terraform-provider-azurerm/pull/25451, https://github.com/hashicorp/terraform-provider-azurerm/pull/25429) need to be merged one by one.

## Change Log

* `azurerm_system_center_virtual_machine_manager_virtual_network` - support this new resource

This is a (please select all that apply):

- [ ] Bug Fix
- [x] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Test Result
--- PASS: TestAccSystemCenterVirtualMachineManagerVirtualNetworkSequential/scvmmVirtualNetwork/basic (337.94s)
--- PASS: TestAccSystemCenterVirtualMachineManagerVirtualNetworkSequential/scvmmVirtualNetwork/requiresImport (316.53s)
--- PASS: TestAccSystemCenterVirtualMachineManagerVirtualNetworkSequential/scvmmVirtualNetwork/complete (324.14s)
--- PASS: TestAccSystemCenterVirtualMachineManagerVirtualNetworkSequential/scvmmVirtualNetwork/update (462.05s)

## References
[Azure doc](https://learn.microsoft.com/en-us/azure/azure-arc/system-center-virtual-machine-manager/overview)
[REST API specs](https://github.com/Azure/azure-rest-api-specs/blob/fe9e014469660ac54057ceacba403c516575c5a1/specification/scvmm/resource-manager/Microsoft.ScVmm/stable/2023-10-07/scvmm.json#L808)

### CODE REVIEW

File: internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_network_resource.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.SystemCenterVirtualMachineManagerVirtualNetworkName,
+		},
+
+		"location": commonschema.Location(),
+
+		"resource_group_name": commonschema.ResourceGroupName(),
```
Comment (ms-zhenhua) [inline internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_network_resource.go:59]: same problem as https://github.com/hashicorp/terraform-provider-azurerm/pull/25429/files#r1547192264

File: internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_network_resource.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			client := metadata.Client.SystemCenterVirtualMachineManager.VirtualNetworks
+
+			var model SystemCenterVirtualMachineManagerVirtualNetworkModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			scvmmServerInventoryItemId, err := inventoryitems.ParseInventoryItemIDInsensitively(model.SystemCenterVirtualMachineManagerServerInventoryItemId)
```
Comment (ms-zhenhua) [inline internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_network_resource.go:85]: same as https://github.com/hashicorp/terraform-provider-azurerm/pull/25429/files#r1547184604

File: internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_network_resource.go [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+					return fmt.Errorf("checking for the presence of an existing %s: %+v", id, err)
+				}
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			parameters := virtualnetworks.VirtualNetwork{
+				ExtendedLocation: virtualnetworks.ExtendedLocation{
+					Type: utils.String("customLocation"),
```
Comment (ms-zhenhua) [inline internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_network_resource.go:104]: same as https://github.com/hashicorp/terraform-provider-azurerm/pull/25429/files#r1547193760

File: internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_network_resource.go [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(*id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := SystemCenterVirtualMachineManagerVirtualNetworkModel{}
+			if model := resp.Model; model != nil {
```
Comment (ms-zhenhua) [inline internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_network_resource.go:146]: same as https://github.com/hashicorp/terraform-provider-azurerm/pull/25429/files#r1547196485

File: internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_network_resource_test.go [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func testAccSystemCenterVirtualMachineManagerVirtualNetwork_update(t *testing.T) {
```
Comment (ms-zhenhua) [inline internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_network_resource_test.go:81]: shall we use the order of basic->complete->update->basic?

**Review by ms-zhenhua — CHANGES_REQUESTED** [review]
Hi @neil-yechenwei ,
Thanks for this PR - I've taken a look through and left some comments inline. If we can fix those up, this should be good to go 👍

File: internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_network_resource.go (hunk 1 above)
Comment (neil-yechenwei) [reply internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_network_resource.go:59]: They can be in different resource group.

File: internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_network_resource.go (hunk 2 above)
Comment (neil-yechenwei) [reply internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_network_resource.go:85]: Updated

File: internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_network_resource.go (hunk 4 above)
Comment (neil-yechenwei) [reply internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_network_resource.go:146]: Updated

File: internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_network_resource_test.go (hunk 5 above)
Comment (neil-yechenwei) [reply internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_network_resource_test.go:81]: Updated

**Comment by neil-yechenwei** [comment]
@ms-zhenhua , thanks for the comments. I updated PR. Please take another look. Thanks,

**Review by ms-zhenhua — APPROVED** [review]
@neil-yechenwei , thank you for your updates. LGTM~

**Comment by neil-yechenwei** [comment]
@katbyte , I noticed that the PR to add stateRefreshFunc has been merged. So I updated this PR to remove the time sleep accordingly. Below is the test result I just now triggered.

```
--- PASS: TestAccSystemCenterVirtualMachineManagerVirtualNetworkSequential/scvmmVirtualNetwork/basic (436.33s)
--- PASS: TestAccSystemCenterVirtualMachineManagerVirtualNetworkSequential/scvmmVirtualNetwork/requiresImport (323.29s)
--- PASS: TestAccSystemCenterVirtualMachineManagerVirtualNetworkSequential/scvmmVirtualNetwork/complete (310.89s)
--- PASS: TestAccSystemCenterVirtualMachineManagerVirtualNetworkSequential/scvmmVirtualNetwork/update (581.43s)
```

**Review by katbyte — APPROVED** [review]
LGTM 🧁

**Pull request merged**